File: educational-system-consumer-admin-9000/src/main/java/edu/hubu/consumeradmin/service/IAnnouncementContentService.java
package edu.hubu.consumeradmin.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.admin.AnnouncementContent;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "admin-provider-service")
public interface IAnnouncementContentService {

    @GetMapping("/announcementContent/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/announcementContent/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "announcementContentId") String orderBy);

    @GetMapping("/announcementContent/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "announcementContentId") String orderBy
    );

    @PostMapping("/announcementContent/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "announcementContentId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.AnnouncementContent announcementContent
    );

    @PostMapping("/announcementContent/")
    CommonResult save(@RequestBody AnnouncementContent announcementContent);

    @PutMapping("/announcementContent/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody AnnouncementContent announcementContent);


    @DeleteMapping("/announcementContent/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/admin/IAnnouncementContentService.kt
package edu.hubu.commons.service.admin

import edu.hubu.commons.model.admin.AnnouncementContent;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface IAnnouncementContentService {
    fun save(t: AnnouncementContent): AnnouncementContent
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: AnnouncementContent): AnnouncementContent
    fun selectOneById(id: String): AnnouncementContent
    fun selectAllOrderBy(order: String = "announcementContentId"): List<AnnouncementContent>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "announcementContentId"): List<AnnouncementContent>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "announcementContentId", t: edu.hubu.commons.mybatis.model.AnnouncementContent): List<edu.hubu.commons.mybatis.model.AnnouncementContent>
}
